### PR TITLE
Add h5md group

### PR DESCRIPTION
**Pull request by pdebuyl**
Fixes #

Description of changes:
 - Add H5MD metadata in H5MD files, according to the specification http://nongnu.org/h5md/h5md.html#h5md-metadata
 - The appropriate version if H5MD 1.1 because at least the addition of connectivity

I tested locally with Python 2 and 3, filing the PR to see CI testing.

PR Checklist
------------
 - [ ] Tests?
   - [ ] Interface
   - [ ] Core 
 - [ ] Docs?

### CODE REVIEW

File: testsuite/h5md.py [hunk 1]
```diff
@@ -69,6 +69,17 @@ def setUpClass(cls):
             os.remove('test.h5')
         cls.py_file = cls.py_pos = cls.py_vel = cls.py_f = cls.py_id = cls.py_img = None
 
+    def test_metadata(self):
+        """Test if the H5MD metadata has been written properly."""
```
Comment (KaiSzuttor) [inline testsuite/h5md.py:73]: please use ```assertEqual```

File: testsuite/h5md.py [hunk 2]
```diff
@@ -69,6 +69,17 @@ def setUpClass(cls):
...
+        self.assertTrue(self.py_file['h5md'].attrs['version'][0] == 1)
+        self.assertTrue(self.py_file['h5md'].attrs['version'][1] == 1)
+        self.assertTrue('creator' in self.py_file['h5md'])
```
Comment (KaiSzuttor) [inline testsuite/h5md.py]: please use ```assertIn```

**Review by KaiSzuttor — CHANGES_REQUESTED** [review]
Thank you for your effort, if you change the asserts this can be merged

**Comment by pdebuyl** [comment]
Hi @KaiSzuttor thanks for the quick feedback :-)

I made the changes and wait for CI.

File: testsuite/h5md.py (hunk 1 above)
Comment (pdebuyl) [reply testsuite/h5md.py:73]: done

File: testsuite/h5md.py (hunk 2 above)
Comment (pdebuyl) [reply testsuite/h5md.py]: also done

**Review by KaiSzuttor — APPROVED** [review]

**Review by fweik — APPROVED** [review]

**Pull request merged**